### ISSUE TITLE
Change encoding profile dependency starting state per project

### ISSUE THREAD

**Issue by saerdnaer** [crs-tools/tracker#157]
As of now sub-tickets require their parent to be released – why not change this to already start with processing after master is in state postprocessed or higher. 

https://github.com/crs-tools/tracker/blob/0d92c8918e3f666ab1ef279b9e12c7ccf53404e4/src/Application/View/tickets/view/_status.html.php#L30-L34

[side question: Is there a way to convert the state enum text values to there integer representation?]

**Comment by a-tze** [comment]
@saerdnaer please provide a description.

**Comment by jjeising** [comment]
> why not change this to already start with processing after master is in state postprocessed or higher.

This was a deliberate decision related to the architecture of the stack when this feature was introduced. At this point the released step copied the files to the internet which was required for access of cloud based encoders.

It might be possible to move this to an earlier state now, but when modifying this I would like to see an option on the parent encoding profile (_enable dependent profiles when in state [select state]_) with released being the default.

Side note: the linked code is only responsible for a view helper, the relevant code is here:

https://github.com/crs-tools/tracker/blob/0d92c8918e3f666ab1ef279b9e12c7ccf53404e4/src/Application/Migrations/22_view_serviceable_tickets.sql#L33

> [side question: Is there a way to convert the state enum text values to there integer representation?]

Why would you want to do that? They already [support ordering in PostgreSQL](https://www.postgresql.org/docs/9.6/static/datatype-enum.html#AEN6923), though getting numeric values [is not impossible entirely](https://stackoverflow.com/questions/12344213/postgresql-is-it-possible-cast-enum-to-integer).

**Comment by a-tze** [comment]
There are multiple issues with a change.

First of all, for the reasons already explained a different "parent minimum state" is only useful as a project property, since the setup may differ from event to event (maybe even additionally to such a setting in an encoding profile). But this would increase the complexity even more, and it's already very complex. Note that this is a code path that is executed every time when a worker node asks for pending tickets.

Second, if you do not wait for the master to be released you may run into race conditions that Publishing currently cannot handle correctly, e.g. having an audio-only-asset as first asset of a lecture to be released. 

To summarize: waiting for master to be released is just the safest option. An improved implementation should be something like "allow subtickets be up to state postprocessed if parent is at least checked(*), allow releasing only if parent is at least released".
Or something like "allow only states up to the parent's state" 🤔 this could be done efficiently but needs some thinking if there are side effects.

**Comment by a-tze** [comment]
@jjeising I created a branch with the necessary changes. In the last commit there is a migration SQL. I would like to remove that commit if we have an idea how to proceed with migrations. Otherwise I will rename the file to something like we had before.
I'm sure the UI change needs some polishing. Feel free to commit to the branch.

**Comment by a-tze** [comment]
Note: I deployed the branch on the voc instance, where the feature is currently tested with a non-time-critical event.

**Comment by a-tze** [comment]
This feature is implemented and tested in 565aa20.
